pilot/pkg/model/telemetry.go

```diff
@@ -1101,6 +1118,8 @@ func generateStatsConfig(class networking.ListenerClass, filterConfig telemetryF
 	cfg := stats.PluginConfig{
 		DisableHostHeaderFallback: disableHostHeaderFallback(class),
 		TcpReportingDuration:      filterConfig.ReportingInterval,
+		RotationInterval:          metricRotationInterval(),
```

Comment:
I would really appreciate more comments - either in code or  where the settings or annotation is defined.

Does this apply to all envoy metrics ?  All Istio metrics ? What are the implications for prometheus or other collectors ? Links to any envoy designs ? Any equivalent feature in OpenTelemetry - since gRPC and non-envoy will use this ? 

Can an equivalent feature be implemented in ztunnel ?   

If we want this to progress to a proper feature/API or be on by default ( which I think we should) - I think we need to  have more info and make sure it is applicable to all data plane implementations.

Reply:
@costinm I have updated PR's description, hope that helps understand

Reply:
> I would really appreciate more comments - either in code or where the settings or annotation is defined.
> 
> Does this apply to all envoy metrics ? All Istio metrics ? What are the implications for prometheus or other collectors ? Links to any envoy designs ? Any equivalent feature in OpenTelemetry - since gRPC and non-envoy will use this ?

Applies to istio metrics only. Envoy metrics don't need this as they are tied to xDS resources, not the data path.
The implementation is https://github.com/istio/proxy/pull/4597 if you want details. @ramaraochavali I think wants the same feature in the core envoy for gRPC, which we should look into, and eventually replace our custom implementation. We will need a more principled telemetry shedding solution anyways, this is mostly a bandaid to stop a trivial way to OOM Istio ingresses.

Metric expiry applies to all sinks, prometheus and opentelemetry included. 

> 
> Can an equivalent feature be implemented in ztunnel ?
> 

Ztunnel needs metric expiry and shedding, like any other telemetry producer.

> If we want this to progress to a proper feature/API or be on by default ( which I think we should) - I think we need to have more info and make sure it is applicable to all data plane implementations.

Why? We're addressing the Envoy issue with its stats system, not some universal API decision to expiry metrics. There's no loss of metrics with this change.